test/CMakeLists.txt

```diff
@@ -21,6 +21,21 @@ set_target_properties(Catch2WithMain PROPERTIES EXPORT_COMPILE_COMMANDS OFF)
 get_target_property(CATCH2_INCLUDE_DIRS Catch2 INTERFACE_INCLUDE_DIRECTORIES)
 target_include_directories(Catch2 SYSTEM INTERFACE ${CATCH2_INCLUDE_DIRS})
 
+# enable precompiled headers
+if (SFML_ENABLE_PCH)
+    message(VERBOSE "enabling PCH for Catch2")
+    target_precompile_headers(Catch2 PRIVATE ${PROJECT_SOURCE_DIR}/src/SFML/PCH.hpp)
+
+    # necessary to find `WindowsHeader.hpp` on Windows
+    target_include_directories(Catch2 PRIVATE ${PROJECT_SOURCE_DIR}/src)
+
+    # necessary to find SFML headers from Catch2 through the PCH header
+    target_include_directories(Catch2 PRIVATE ${PROJECT_SOURCE_DIR}/include)
+
+    # necessary to find Catch2 headers from SFML through the PCH header
+    target_include_directories(sfml-system PRIVATE ${CATCH2_INCLUDE_DIRS})
```

Comment:
@eXpl0it3r: this seems to solve the build on Windows, if you have any suggestions on a more elegant solution I'm all ears :)

Reply:
I wonder if there isn't some include directory variable thingy from the target we can use. Maybe @ChrisThrasher has an idea.